src/main/java/salesforce/ui/pages/assets/NewAssetPage.java

```diff
@@ -72,89 +73,137 @@
     @FindBy(xpath = "//button[@title=\'Save\']")
     private WebElement saveBtn;
 
+    private static final String COMMON_LOCATOR = "//label/span[text()='%s']/../..//input";
+    private static final String DESCRIPTION_LOCATOR = "//label/span[text()='Description']/../..//textarea";
+    private static final String SELECT_LOCATOR = "//div[@role='listbox']//a[.//div[text()='%s']]";
+    private static final String BUTTON_LOCATOR = "//button[@title='%s']";
+
     @Override
     protected void waitForPageLoaded() {
         wait.until(ExpectedConditions.visibilityOf(saveBtn));
+        wait.until(ExpectedConditions.visibilityOf(userNameTxtBox));
     }
 
-    public CreateAssetPage setUserName(final String userName) {
+    public NewAssetPage setUserName(final String userName) {
         webElementAction.setInputFields(userNameTxtBox, userName);
         return this;
     }
 
-    public CreateAssetPage setSerialNumber(final String serialNumber) {
+    public NewAssetPage setField(final String field, final String value) {
+        webElementAction.setInputFields(driver.findElement(By.xpath(String.format(COMMON_LOCATOR, field))), value);
+        return this;
+    }
+
+    public NewAssetPage clickField(final String field) {
+        webElementAction.clickFields(driver.findElement(By.xpath(String.format(COMMON_LOCATOR, field))));
+        return this;
+    }
+
+    /**
+     * Selects a webElement and a inside one.
+     *
+     * @param field is first webElement that will be clicked.
+     * @param optionField is second webElement that will be clicked.
+     * @return a NewAssetPage entity.
+     */
+    public NewAssetPage clickOptionField(final String field, final String optionField) {
+        webElementAction.clickOptionField(driver.findElement(By.xpath(String.format(COMMON_LOCATOR, field))),
+                driver.findElement(By.xpath(String.format(SELECT_LOCATOR, optionField))));
+        return this;
+    }
+
+    public NewAssetPage selectField(final String field) {
```

Comment:
each method should have its own javadoc.

Reply:
You were right. I already added them.